common/lib/xmodule/xmodule/modulestore/xml.py

```diff
@@ -504,23 +526,40 @@ def get_policy(usage_id):
             if isinstance(course_descriptor, ErrorDescriptor):
                 return course_descriptor
 
-            # NOTE: The descriptors end up loading somewhat bottom up, which
-            # breaks metadata inheritance via get_children().  Instead
-            # (actually, in addition to, for now), we do a final inheritance pass
-            # after we have the course descriptor.
-            compute_inherited_metadata(course_descriptor)
-
-            # now import all pieces of course_info which is expected to be stored
-            # in <content_dir>/info or <content_dir>/info/<url_name>
-            self.load_extra_content(system, course_descriptor, 'course_info', self.data_dir / course_dir / 'info', course_dir, url_name)
+            if not library:
+                # NOTE: The descriptors end up loading somewhat bottom up, which
+                # breaks metadata inheritance via get_children().  Instead
+                # (actually, in addition to, for now), we do a final inheritance pass
+                # after we have the course descriptor.
+                compute_inherited_metadata(course_descriptor)
```

Comment:
The current plan for XBlocks is to have field data be accessed via the `field-data` service, and thus have a single `FieldData` shared across all of the blocks in a course. This is how the XBlock xml parsing code works. In contrast, XModules expect to have one `FieldData` per XModule.

I would love to move all of the inheritance to `InheritingFieldData` instances, which simply do XBlock tree traversal on demand, rather than requiring inheritance to be computed downwards. In fact, for XML modulestores (and thus XML import), switching everything over to `InheritingFieldData` should be quite feasible, since the blocks are always in memory already.